### PR TITLE
Address review comments for PR 6118, 6169, 6175

### LINKED ISSUE #6174: Address review comments for PR 6118, 6169, 6175
https://github.com/getodk/collect/pull/6118:

- [ ] https://github.com/getodk/collect/pull/6118#discussion_r1618797076

https://github.com/getodk/collect/pull/6169:

- [x] https://github.com/getodk/collect/pull/6169#discussion_r1630983497
- [x] https://github.com/getodk/collect/pull/6169#discussion_r1630985934
- [x] https://github.com/getodk/collect/pull/6169#discussion_r1630987312
- [x] https://github.com/getodk/collect/pull/6169#discussion_r1631095161
- [x] https://github.com/getodk/collect/pull/6169#discussion_r1631096623
- [x] https://github.com/getodk/collect/pull/6169#discussion_r1631106898
- [x] https://github.com/getodk/collect/pull/6169#discussion_r1631109113
- [ ] https://github.com/getodk/collect/pull/6169#discussion_r1629622207

#6175 

- [x] https://github.com/getodk/collect/pull/6175#discussion_r1636317407
- [x] https://github.com/getodk/collect/pull/6175#discussion_r1636468303

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #6174 

#### Why is this the best possible solution? Were any other approaches considered?
I was meant to handle this earlier, but I got caught up with other tasks and kept postponing it. Better late than never. I've addressed most of the comments, though not all of them (some were outdated and not relevant anymore). Some are less critical and/or too risky, as they would require retesting the functionality so I skipped them. I think what I've fixed is enough and we can merge the changes and close the issue.
The comments I skipped are:
- https://github.com/getodk/collect/pull/6118#discussion_r1618797076: I skipped this one because it doesn't seem that important. It doesn’t offer a significant advantage, so taking the risk to add such a refactoring doesn’t make much sense. Also, the code has changed significantly since the comment was added.
- https://github.com/getodk/collect/pull/6169#discussion_r1629622207: Testing this is not straightforward, so it might be better to file a separate issue if it’s considered important. We also have a similar method elsewhere, so a broader refactoring might be necessary.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
I think it doesn't require testing.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] added any new strings with date formatting to `DateFormatsTest`
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: test-shared/src/main/java/org/odk/collect/testshared/Interactions.kt [hunk 1]
```diff
@@ -57,4 +59,14 @@ object Interactions {
         onView(view).perform(ViewActions.replaceText(text))
         closeSoftKeyboard()
     }
+
+    fun onListItem(listId: Int, position: Int, childViewId: Int): ViewInteraction {
```
Comment (seadowg) [inline test-shared/src/main/java/org/odk/collect/testshared/Interactions.kt]: This doesn't really fit the pattern of the other helpers in `Interactions`/`Assertions`. Maybe instead you could pull out a custom helper (maybe to `ViewMatchers`) like `atPositionInRecyclerView(listId: Int, position: Int, childViewId: Int): Matcher<View` that you can then use with our helpers like `clickOn`:

```kotlin
clickOn(atPositionInRecyclerView(R.id.layers, 1, R.id.path))
```

**Review by seadowg — CHANGES_REQUESTED** [review]

File: test-shared/src/main/java/org/odk/collect/testshared/Interactions.kt (hunk 1 above)
Comment (grzesiek2010) [reply test-shared/src/main/java/org/odk/collect/testshared/Interactions.kt]: Done.

**Review by seadowg — APPROVED** [review]

**Comment by seadowg** [comment]
I've added "needs testing" to this as it derisks releasing it as part of the next hotfix. The main thing to check here is just a general run through of offline layer importing/management. There's no behaviour changes - just code and test improvements.

**Comment by srujner** [comment]
**Tested with Success!**

Verified on Device with Android 10,14

Verified cases:
- Importing/Managing Offline layers; 
- Adding/Deleting Layers from Select Offline Layers panel and from Settings;
- Regression check on forms with Geo Questions;
- Select One from map; 
- Different map sources.

**Pull request merged**